sandbox/extract-imagery-metadata_mg.R

```diff
@@ -0,0 +1,181 @@
+
+
+# --- 2. Example of how to use this sandbox script to write a metadata extraction function ---
+
+# Select an EXIF file to test on, and prep the EXIF data by loading it as a geospatial data frame
+# using the 'prep_exif' function. The 'prep_exif' function is defined in
+# 'R/imagery-metadata-extraction_general.R'            # nolint
+exif_file = exif_files[1]
+exif = prep_exif(exif_file)
+# Note that the 'prep_exif' function returns the EXIF data as a geospatial data frame (an 'sf'
+# object) with point geometry. So any geospatial operations you attempt on it should use functions
+# from the 'sf' package. If you are more familiar with 'terra' objects and would rather work with
+# them, let Derek know and we can create an option to return 'terra::vect' objects.
+
+# Between the BEGIN and END comments below, write code to extract the metadata attribute you're
+# working on. When you're done, you can wrap it in a function definition, taking only one parameter,
+# 'exif'. Here is an example of developing code to extract the number of images in an imagery
+# dataset.
+
+# BEGIN FUNCTION CODE
+
+# Get the number of images in the dataset by counting the rows of the EXIF data frame
+image_count = nrow(exif)
+
+# END FUNCTION CODE
+
+
+# Now here is an example of turning that code into a function
+
+extract_image_count = function(exif) {
+
+  # Get the number of images in the dataset by counting the rows of the EXIF data frame
+  image_count = nrow(exif)
+
+  return(image_count)
+
+}
+
+
+# Now you can test the function on the EXIF data
+
+image_count = extract_image_count(exif)
+image_count
+
+# Once it is working right, you can move this function to your
+# 'R/imagery-metadata-extraction_<initials>.R' file and then add a call to this function from within
+# your 'extract_metadata_<initials>' function. Once it is in there, then you can run the top part of
+# this script again, and when it extracts the metadata for each EXIF dataset, your additional
+# metadata should be included.
+
+
+#START CODE FOR CENTROID LAT/LONG
+
+GPSLocation = exif$GPSPosition
+
+#splitting the list (was this step really necessary for my dataset? I'm not really sure this actually helped?)
+coordinates <- strsplit(GPSLocation, split = ";")
+
+# Separating the GPS from the whole exif file
+almostdone <- strsplit(unlist(coordinates), split = ",")
+
+#Creating a seperate dataframe with just the coordinates to call upon later
+isolatedcoordinates <- as.data.frame(do.call(rbind, almostdone))
+
+library(stringr)
+#Need to download this package as it will help with the following functions
+separator(df)
+
+install.packages("tidyverse")
+library(tidyverse)
+
+df2 = separate(exif,
+        col = GPSPosition,
+        into = c("lat", "long"),
+        sep = " " )
+#Here this worked, instead of doing by number of characters, did it by seperating at
+#the space in between the two (since the number of characters differed from row to row).
```

Comment:
Nice work getting lat/lon this way! However, because `exif` is a spatial object, it is slightly preferable to use the following approach:

```
coords = sf::st_coordinates(exif)
lat = coords[, 1]
lon = coords[, 2]
```
This will make sure your code (at least this part of it) is robust to future drone models that might have EXIF columns named differently. You can trust that whatever the right column names were, they were already used for obtaining the lat/lon that is embedded in the `exif` spatial object.

Reply:
Even better, you would compute the centroid as a geospatial `sf` object, which you could do by using some functions in the `sf` package applied to the `exif` object, without ever having to extract coordinates. See if you can find any suggestions for this workflow online, and if not I can hep with it.

Reply:
Agree with @youngdjn here about calculating the centroid directly, then passing that to the suntools::solarnoon() calculation. It does look like it accepts {sf} objects (https://github.com/adokter/suntools/blob/374fff7b2f595fabb8bf75d928c63b6e2f099e98/R/solarnoon.R#L16-L42). Nice find on this package!

Note for posterity that latitude is actually the 2nd column in the result of `sf::st_coordinates(exif)`, not the first. So Derek's code would be: 
```
coords = sf::st_coordinates(exif)
lon = coords[, 1]
lat = coords[, 2]
```

Reply:
Oops I always do that! Updating my comment to minimize future confusion.